src/NuGet.Core/NuGet.ProjectManagement/LoggerAdapter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using NuGet.Logging;
+
+namespace NuGet.ProjectManagement
+{
+    /// <summary>
+    /// ILogger -> INuGetProjectContext
+    /// </summary>
+    public class LoggerAdapter : ILogger
+    {
+        public INuGetProjectContext ProjectLogger { get; }
```

Comment:
Or it could just carry a logger. INuGetProjectContext has kept from NuGet.Core and is somewhat confusing as a logger itself.